src/include/common/task_system/task.h

```diff
@@ -91,7 +91,7 @@ class Task {
         children; // Dependency tasks that needs to be executed first.
 
 protected:
-    std::mutex mtx;
+    std::mutex taskMtx;
```

Comment:
Because there is code in task_scheduler.cpp that needs to grab taskMtx vs task queue's mtx (which I called taskSchedulerMtx) and giving them clearer names makes the code clearer to understand. This part of the code is very important because of the concurrency guarantees we need to satisfy for correct implementation of dependencies and which writes will be globally visible to all threads.